distributed/batched.py

```diff
@@ -152,17 +155,20 @@ def close(self, timeout=None):
             return
         self.please_stop = True
         self.waker.set()
-        yield self.stopped.wait(timeout=timeout)
+        await asyncio.wait_for(
+            self.stopped.wait(), timeout=timeout.total_seconds() if timeout else None
+        )
         if not self.comm.closed():
             try:
                 if self.buffer:
                     self.buffer, payload = [], self.buffer
-                    yield self.comm.write(
+                    await self.comm.write(
                         payload, serializers=self.serializers, on_error="raise"
                     )
             except CommClosedError:
                 pass
-            yield self.comm.close()
+            await self.comm.close()
+        await self._background_task
```

Comment:
waiting for `self.stopped.wait()` should only finish once `self._background_task` has - so this await should be redundant